components/eamxx/src/physics/mam/eamxx_mam_aci_process_interface.cpp

```diff
@@ -0,0 +1,1299 @@
+  MAMAci::const_view_2d zm    = dry_atmosphere.z_mid;
+  MAMAci::const_view_2d pdel  = dry_atmosphere.p_del;
+  MAMAci::const_view_2d p_int = dry_atmosphere.p_int;
+  MAMAci::const_view_2d nc    = dry_atmosphere.nc;
+
+  //----------------------------------------------------------------------
+  // ## Declare local variables for class variables
+  //(FIXME: GPU hack, revisit this)
+  //----------------------------------------------------------------------
+  MAMAci::view_2d loc_raercol_cw[mam4::ndrop::pver][2];
+  MAMAci::view_2d loc_raercol[mam4::ndrop::pver][2];
+  MAMAci::view_2d loc_qqcw[mam4::ndrop::ncnst_tot];
+  MAMAci::view_2d loc_ptend_q[mam4::aero_model::pcnst];
+  MAMAci::view_2d loc_coltend[mam4::ndrop::ncnst_tot];
+  MAMAci::view_2d loc_coltend_cw[mam4::ndrop::ncnst_tot];
+
+  for(int i = 0; i < mam4::ndrop::pver; ++i) {
+    for(int j = 0; j < 2; ++j) {
+      loc_raercol_cw[i][j] = raercol_cw[i][j];
+      loc_raercol[i][j]    = raercol[i][j];
+    }
+  }
+
+  for(int i = 0; i < mam4::ndrop::ncnst_tot; ++i) {
+    loc_coltend[i]    = coltend[i];
+    loc_coltend_cw[i] = coltend_cw[i];
+  }
+
+  for(int i = 0; i < mam4::aero_model::pcnst; ++i) loc_ptend_q[i] = ptend_q[i];
+
+  MAMAci::view_2d qqcw_fld_work_loc[25];
+  for(int i = 0; i < mam4::ndrop::ncnst_tot; ++i)
+    qqcw_fld_work_loc[i] = qqcw_fld_work[i];
+
+  MAMAci::view_3d state_q_work_loc = state_q_work;
+
+  //----------------------------------------------------------------------
+  // ## Assign scratch memory for work variables
+  //----------------------------------------------------------------------
+
+  MAMAci::view_2d eddy_diff    = dropmixnuc_scratch_mem[0];
+  MAMAci::view_2d zn           = dropmixnuc_scratch_mem[1];
+  MAMAci::view_2d csbot        = dropmixnuc_scratch_mem[2];
+  MAMAci::view_2d zs           = dropmixnuc_scratch_mem[3];
+  MAMAci::view_2d overlapp     = dropmixnuc_scratch_mem[4];
+  MAMAci::view_2d overlapm     = dropmixnuc_scratch_mem[5];
+  MAMAci::view_2d eddy_diff_kp = dropmixnuc_scratch_mem[6];
+  MAMAci::view_2d eddy_diff_km = dropmixnuc_scratch_mem[7];
+  MAMAci::view_2d qncld        = dropmixnuc_scratch_mem[8];
+  MAMAci::view_2d srcn         = dropmixnuc_scratch_mem[9];
+  MAMAci::view_2d source       = dropmixnuc_scratch_mem[10];
+  MAMAci::view_2d dz           = dropmixnuc_scratch_mem[11];
+  MAMAci::view_2d csbot_cscen  = dropmixnuc_scratch_mem[12];
+  MAMAci::view_2d raertend     = dropmixnuc_scratch_mem[13];
+  MAMAci::view_2d qqcwtend     = dropmixnuc_scratch_mem[14];
+
+  //---------------------------------------------------------------------------
+  // ## Initialize the ndrop class.
+  //---------------------------------------------------------------------------
+  const int ntot_amode        = mam_coupling::num_aero_modes();
+  const int maxd_aspectype    = mam4::ndrop::maxd_aspectype;
+  const int nspec_max         = mam4::ndrop::nspec_max;
+  int nspec_amode[ntot_amode] = {};
```

Comment:
We may need to move the computation of these parameter inside of the parallel_for.

Reply:
I tested it on both GPUs and CPUs. I get the right values whether these are in or out of the Kokkos loop. As we discussed, I am planning to keep the code as is for now.